File: backend/src/main/java/com/example/backend/model/response/ReservationResponse.java
package com.example.backend.model.response;

import com.example.backend.model.TicketType;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class ReservationResponse {
    private String name;
    private String surname;
    private String movieTitle;
    private Integer screeningRoomNumber;
    private LocalDateTime startTime;
    private BigDecimal totalPrice;
    private List<Ticket> tickets;
    private LocalDateTime expirationTime;

    @Builder
    @AllArgsConstructor
    @NoArgsConstructor
    @Getter
    @Setter
    public static class Ticket {
        private TicketType ticketType;
        private Integer row;
        private Integer seatNumber;
        private BigDecimal price;
    }

}

File: frontend/src/app/models/reservation_response.ts
export interface ReservationResponse {
    name: string;
    surname: string;
    movieTitle: string;
    screeningRoomNumber: number;
    startTime: string;
    totalPrice: number;
    tickets: TicketResponse[];
    expirationTime: string;
}

export interface TicketResponse {
    ticketType: string;
    row: number;
    seatNumber: number;
    price: number;
}